src/hls.ts

```diff
@@ -284,8 +284,7 @@ export default class Hls implements HlsEventEmitter {
     this.url = null;
     if (this.networkControllers) {
       this.networkControllers.forEach((component) => component.destroy());
-      // @ts-ignore
-      this.networkControllers = null;
+      this.networkControllers = undefined;
```

Comment:
What if we just did this?

```suggestion
      this.networkControllers.length = 0;
```

Reply:
That's a much simpler fix.  :)  I made the same change to `coreComponents` below, too.

Reply:
Thanks!